hadoop-yarn-project/hadoop-yarn/hadoop-yarn-common/src/main/java/org/apache/hadoop/yarn/client/RMProxy.java

```diff
@@ -99,7 +99,8 @@ protected static <T> T createRMProxy(final Configuration configuration,
         ? (YarnConfiguration) configuration
         : new YarnConfiguration(configuration);
     RetryPolicy retryPolicy = createRetryPolicy(conf,
-        (HAUtil.isHAEnabled(conf) || HAUtil.isFederationFailoverEnabled(conf)));
+        (HAUtil.isHAEnabled(conf) || (HAUtil.isFederationEnabled(conf)
```

Comment:
This boolean is not easy to read.
Can we extract with some nice comment and reuse it?
Something like:
```
/**
 * Explain the logic
 */
private static boolean isFailoveEnabled(conf) {
  if (HAUtil.isHAEnabled(conf)) {
    // Comment explaining why
    return true;
  }
  if (HAUtil.isFederationEnabled(conf) && HAUtil.isFederationFailoverEnabled(conf)) {
    // Comment explaining why
    return true;
  }
  return false;
}
```

Reply:
Thanks @goiri for reviewing the PR and sharing your comments. I have address your comments in my last commit - Can you please review it?